core/src/autogluon/core/utils/utils.py

```diff
@@ -260,7 +260,10 @@ def get_pred_from_proba_df(y_pred_proba, problem_type=BINARY):
 
 def get_pred_from_proba(y_pred_proba, problem_type=BINARY):
     if problem_type == BINARY:
-        y_pred = [1 if pred >= 0.5 else 0 for pred in y_pred_proba]
+        # Using > instead of >= to align with Pandas `.idxmax` logic which picks the left-most column during ties.
+        # If this is not done, then predictions can be inconsistent when converting in binary classification from multiclass-form pred_proba and
+        # binary-form pred_proba when the pred_proba is 0.5 for positive and negative classes.
+        y_pred = [1 if pred > 0.5 else 0 for pred in y_pred_proba]
```

Comment:
Shall we move `0.5` into a property or method method call? Rationale: at some point we'll add optimization of the threshold so having either method or named property would be a natural place to change instead of looking for magic value here.

Reply:
Yes, but this should be a separate PR as it is a major change. I've had this in mind for awhile